src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/SqlQueryGenerator.cs

```diff
@@ -81,18 +82,25 @@ public override object VisitSqlRoot(SqlRootExpression expression, SearchOptions
                     throw new InvalidOperationException("Expected no predicates on the Resource table because of the presence of TableExpressions");
                 }
 
+                // Union All expressions must be executed first than all other expressions. The overral idea is that Union All expressions will
+                // filter the highest group of records, and the following expressions will be executed on top of this group of records.
```

Comment:
Is it possible to have something like `UnionAll (Condition1, Condition2(UnionAll(Condition2, Condition2b))`? i.e. if there it can be nested somewhere else lower in the tree

Reply:
I can't think of one since I think most sub expressions will probably my `IN` or `OR`, but if its not supported might make sense to throw an exception in that case

Reply:
More for future if we do come up with a scenario